### ISSUE TITLE
NoSuchElementException during memory trimming

### ISSUE THREAD

**Issue by fillobotto** [bumptech/glide#2562]
**Glide Version**: 4.1.1

**Integration libraries**: -

**Device/Android Version**: Moto E with 4G LTE (2nd Gen) Android 6.0

**Issue details / Repro steps / Use case background**: 
I cannot reproduce the issue, it just happened on a device out of 5000 users, more or less. RAM usage at the time of crashes was not critical.

**Glide load line / `GlideModule` (if any) / list Adapter code (if any)**:
I'm using a custom Glide module:
```
@GlideModule
public class CustomGlideModule extends AppGlideModule {
    @Override
    public void applyOptions(Context context, GlideBuilder builder) {
        MemorySizeCalculator calculator = new MemorySizeCalculator.Builder(context)
                .setMemoryCacheScreens(1)
                .setLowMemoryMaxSizeMultiplier(0.2f)
                .setMaxSizeMultiplier(0.2f)
                .build();
        builder.setMemoryCache(new LruResourceCache((int) (calculator.getMemoryCacheSize() * 0.8)));
        builder.setDiskCache(new ExternalCacheDiskCacheFactory(context));
        builder.setBitmapPool(new LruBitmapPool((int) (calculator.getBitmapPoolSize() * 0.6)));
        builder.setDefaultRequestOptions(new RequestOptions().fitCenter().placeholder(R.drawable.ic_music).format(DecodeFormat.PREFER_RGB_565).sizeMultiplier(0.7f));
    }

    @Override
    public boolean isManifestParsingEnabled() {
        return false;
    }
}
```

Images are loaded in many ways all around the app, mostly like that:
```
Glide.with(Activity or Fragment)
                    .load(io.File or Uri)
                    .apply(new RequestOptions())
                    .thumbnail(0.2f)
                    .into(ImageView);
```

**Stack trace / LogCat**:
```
Fatal Exception: java.util.NoSuchElementException
       at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:349)
       at java.util.LinkedHashMap$EntryIterator.next(LinkedHashMap.java:375)
       at java.util.LinkedHashMap$EntryIterator.next(LinkedHashMap.java:375)
       at com.bumptech.glide.util.LruCache.trimToSize(SourceFile:170)
       at com.bumptech.glide.load.engine.cache.LruResourceCache.trimMemory(SourceFile:50)
       at com.bumptech.glide.Glide.trimMemory(SourceFile:478)
       at com.bumptech.glide.Glide.onTrimMemory(SourceFile:669)
       at android.app.Application.onTrimMemory(Application.java:134)
       at android.app.ActivityThread.handleTrimMemory(ActivityThread.java:4413)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1542)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5443)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
```

**Comment by sjudd** [comment]
This will be fixed by #2550, but it doesn't explain why the cache is out of sync.

Are you using any custom decoders, or Resource classes? Are you always loading Bitmaps? Or do you load animated Drawables or resources as well?

**Comment by fillobotto** [comment]
Everything custom I have is that Glide module. I load images from File but also from urls and finally from drawable as placeholder. Images are jpeg and png only.
